Source/Application/Network/ImageProvider.swift

```diff
@@ -0,0 +1,25 @@
+import UIKit
+
+struct ImageProvider {
+
+    private let requestServicing: RequestServicing
+
+    init(requestServicing: RequestServicing) {
+        self.requestServicing = requestServicing
+    }
+
+    func withImage(at url: URL, execute closure: @escaping (Result<UIImage, AnyError>) -> Void) {
+        self.requestServicing.request(via: .get, url: url, queryItems: nil, headerItems: nil) { result in
+            switch result {
```

Comment:
Speaking to the previous comment about accidentally introducing a new state, I think that this is a better representation of capturing our states in easy-to-read blocks for developers on the team.